pkg/cluster/compute/affinityrule/affinityrule.go

```diff
@@ -97,3 +98,45 @@ func GetAffinityRule(whost *host.WmiHost, affinityRuleName string) (caffinityRul
 	caffinityRule = &AffinityRule{wmiafRule}
 	return
 }
+
+// SetAffinityRule updates an existing affinity rule
+func SetAffinityRule(whost *host.WmiHost, name string, ruleType int, softAntiAffinity bool) (err error) {
+	affinityRule, err := GetAffinityRule(whost, name)
+	if err != nil {
+		return
+	}
+	defer affinityRule.Close()
+
+	propeties := affinityRule.GetClass().GetPropertiesNames()
+	if len(propeties) == 0 {
+		err = errors.Wrapf(errors.Unknown, "Failed to get the properties of MSCluster_AffinityRule")
+		return
+	}
+
+	softAntiAffinitySupported := false
```

Comment:
do this only for rule=anti-affinity, softAntiAffinity=true?

Reply:
Fixed in new patch